Source/Spatializer/CrossPlatform/HrtfWrapper.cpp

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "HelperMacros.h"
+#include "HrtfWrapper.h"
+#include <exception>
+#include <cstring>
+
+// Statics
+std::unique_ptr<HrtfWrapper> HrtfWrapper::s_HrtfWrapper;
+
+HrtfWrapper::SourceInfo::SourceInfo(uint32_t index, HrtfInputBuffer* const sourceBuffer)
+    : m_SourceIndex(index), m_SourceBuffer(sourceBuffer)
+{
+}
+
+HrtfWrapper::SourceInfo::~SourceInfo()
+{
+    m_SourceBuffer->Buffer = nullptr;
+    m_SourceBuffer->Length = 0;
+    if (HrtfWrapper::s_HrtfWrapper)
+    {
+        s_HrtfWrapper->ReleaseSource(m_SourceIndex);
+    }
+}
+
+void HrtfWrapper::InitWrapper()
+{
+    RETURN_RESULT_IF_NULL(HrtfWrapper::s_HrtfWrapper);
```

Comment:
First, this macro breaks my head.

Second, I think this is wrong. How does the wrapper ever get initialized if it always returns if null?

Reply:
Ok, I removed the macros. I miss WIL style macros which IMO improve readability but in this codebase the utility is minimal.

Reply:
I only liked WIL for the HRESULT handling. Anything else I prefer being more verbose (especially in cross-plat OSS code where I assume most people haven't seen WIL?).

Reply:
IMO hresult handling was the least important aspect of WIL. :) A lot of the helper functionality in WIL improved readability, provide consistent mechanisms for exception handling, RAII resource usage (for some of the pesky Windows types), and brought core business logic into focus. But, we don't have to agree, at least right now. ;)